docs/reference/components/sources/aws_s3.cue

```diff
@@ -0,0 +1,225 @@
+						to publish to this queue for the following events:
+
+						- PUT
+						- POST
+						- COPY
+						- Multipart upload completed
+
+						These represent object creation events.
+						""",
+				]
+			}
+		}
+	}
+
+	classes: {
+		commonly_used: true
+		deployment_roles: ["aggregator"]
+		delivery:      "at_least_once"
+		development:   "beta"
+		egress_method: "stream"
+	}
+
+	support: {
+		platforms: {
+			"aarch64-unknown-linux-gnu":  true
+			"aarch64-unknown-linux-musl": true
+			"x86_64-apple-darwin":        true
+			"x86_64-pc-windows-msv":      true
+			"x86_64-unknown-linux-gnu":   true
+			"x86_64-unknown-linux-musl":  true
+		}
+
+		requirements: []
+		warnings: []
+		notices: []
+	}
+
+	configuration: {
+		strategy: {
+			common:      true
+			description: "The strategy to use to consume objects from AWS S3."
+			required:    false
+			type: string: {
+				default: "sqs"
+				enum: {
+					sqs: "Consume S3 objects by polling for bucket notifications sent to an [SQS queue][urls.sqs]."
+				}
+			}
+		}
+		compression: {
+			common:      false
+			description: "The compression format of the S3 objects.."
+			required:    false
+			type: string: {
+				default: "text"
+				enum: {
+					auto: "Vector will try to determine the compression format of the object from its: `Content-Encoding` metadata, `Content-Type` metadata, and key suffix (e.g. `.gz`). It will fallback to 'none' if it cannot determine the compression."
+					gzip: "GZIP format."
+					zstd: "ZSTD format."
+					none: "Uncompressed."
+				}
+			}
+		}
+		sqs: {
+			common:      true
+			description: "SQS strategy options. Required if strategy=`sqs`."
+			required:    false
+			warnings: []
+			type: object: {
+				examples: []
+				options: {
+					poll_secs: {
+						common:      true
+						description: "How often to poll the queue for new messages in seconds."
+						required:    false
+						warnings: []
+						type: uint: {
+							default: 15
+							unit:    "seconds"
+						}
+					}
+					visibility_timeout_secs: {
+						common:      false
+						description: "The visibility timeout to use for messages in secords. This controls how long a message is left unavailable when a Vector receives it. If Vector does not delete the message before the timeout expires, it will be made reavailable for another consumer; this can happen if, for example, the `vector` process crashes."
+						required:    false
+						warnings: ["Should be set higher than the length of time it takes to process an individual message to avoid that message being reprocessed."]
+						type: uint: {
+							default: 300
+							unit:    "seconds"
+						}
+					}
+					delete_message: {
+						common:      true
+						description: "Whether to delete the message once Vector processes it. It can be useful to set this to `false` to debug or during initial Vector setup."
+						required:    false
+						warnings: []
+						type: bool: default: true
+					}
+					queue_name: {
```

Comment:
This one is missing the `common` flag

Reply:
I actually discovered that `required` cannot be used with `common` in our cue schema. I guess the thinking is that `required` implies `common`.

Reply:
Ah I see. I believe I saw a few others that had `required` and `common` defined though?

Reply:
Sorry, I mean if `required` is `true`. If it is `false`, `common` is required to be defined (if I remember correctly).

Reply:
Ah yes, sorry, I had checked that, but I somehow thought I had seen a required `true` and `common` used together, but I was wrong!